environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -238,6 +242,13 @@ private void fireContainerInitializedEvent() {
      */
     public Event<Object> event() {
         checkState();
+        if (event == null) {
+            synchronized (this) {
```

Comment:
We want to be sure the `event` is only initialized once and visible to all threads, that's the reason why it's volatile and initialized in a synchronized block.